src/app/components/ConsensusTransactionMethod/index.tsx

```diff
@@ -75,19 +99,93 @@ export const MethodIcon: FC<MethodIconProps> = ({
       >
         {cloneElement(icon, { sx: { fontSize: Math.ceil(size * iconRatio) } })}
       </Box>
-      {label && <Typography sx={{ textTransform: 'capitalize' }}>{label}</Typography>}
+      {label && (
+        <Typography
+          sx={{
+            fontWeight: 'inherit',
+            textTransform: 'capitalize',
+            order: reverseLabel ? -1 : 0,
+            ...(truncate
+              ? {
+                  minWidth: '100px',
+                  maxWidth: '1px',
+                  whiteSpace: 'nowrap',
+                  overflow: 'hidden',
+                  textOverflow: 'ellipsis',
+                }
+              : {}),
+          }}
+        >
+          {label}
+        </Typography>
+      )}
     </Box>
   )
 }
 
-const getConsensusTransactionMethod = (t: TFunction, method: ConsensusTxMethod | undefined) => {
+const MethodIconWithTruncatedLabel: FC<MethodIconProps> = props => {
+  const elementRef = useRef<HTMLDivElement>(null)
+  const [truncate, setTruncate] = useState(false)
+  const [baseLabelWidth, setBaseLabelWidth] = useState(0)
+  const applyTruncate = useCallback(() => {
+    if (elementRef.current) {
+      const elementWidth = elementRef.current.offsetWidth
+      const parentWidth = (elementRef.current.parentNode as HTMLElement)?.offsetWidth
+      setTruncate(elementWidth === parentWidth || baseLabelWidth > parentWidth)
+    }
+  }, [elementRef, baseLabelWidth])
+
+  useEffect(() => {
```

Comment:
I would prefer useLayoutEffect over useEffect.

Reply:
That just seems less efficient. Why?

Reply:
Yeah, so just because of the element ref. The React docs have great example with a tooltip: https://react.dev/reference/react/useLayoutEffect#measuring-layout-before-the-browser-repaints-the-screen